src/App.tsx

```diff
@@ -1,73 +1,91 @@
-import React from 'react';
+import React, { useMemo, useRef, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { DropdownMenu } from './components/DropdownMenu/DropdownMenu';
+import classNames from 'classnames';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [people] = useState<Person[]>(peopleFromServer);
+  const [query, setQuery] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+  const [normalizedQuery, setNormalizedQuery] = useState('');
+  const timerId = useRef(0);
+  const field = useRef<HTMLInputElement>(null);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const saveQuery = (newQuery: string) => {
+    setQuery(newQuery);
+
+    window.clearTimeout(timerId.current);
+
+    timerId.current = window.setTimeout(() => {
+      setNormalizedQuery(newQuery.trim().toLowerCase());
+
+      if (newQuery !== normalizedQuery) {
```

Comment:
The condition `if (newQuery !== normalizedQuery)` may not work as intended because `normalizedQuery` is updated asynchronously after the timeout. Consider comparing `newQuery.trim().toLowerCase()` directly with `normalizedQuery` to ensure the condition reflects the intended logic.